File: Remove Duplicates from Sorted List.cpp
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
     ListNode* another=head;
     if(head!=NULL) {
       while( another->next!=NULL){
         if (another->val==another->next->val) 
         {
             another->next=another->next->next;
             
         }
         else
         another=another->next;
       }
     }
    return head;
}
    
};

File: first-phase/week2/remove-duplicates-from-sorted-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if head == None:
            return None
        left = head
        right = head.next                                      
        while right and left:
            if right.val != left.val:
                left.next = right
                left = left.next
            right = right.next
        if left:
            left.next = None
        return head